packages/dids/src/methods/did-dht.ts

```diff
@@ -0,0 +1,1568 @@
+
+      // Add a TXT record for the verification method.
+      dnsAnswerRecords.push({
+        type : 'TXT',
+        name : `_${dnsRecordId}._did.`,
+        ttl  : DNS_RECORD_TTL,
+        data : txtData.join(PROPERTY_SEPARATOR)
+      });
+    }
+
+    // Add DNS TXT records for each service.
+    didDocument.service?.forEach((service, index) => {
+      const dnsRecordId = `s${index}`;
+      serviceIds.push(dnsRecordId);
+      const serviceId = service.id.split('#').pop()!; // Remove fragment prefix, if any.
+      const serviceEndpoint = Array.isArray(service.serviceEndpoint)
+        ? service.serviceEndpoint.join(',')
+        : service.serviceEndpoint;
+
+      // Define the data for the DNS TXT record.
+      const txtData = [`id=${serviceId}`, `t=${service.type}`, `se=${serviceEndpoint}`];
+
+      // Add a TXT record for the verification method.
+      dnsAnswerRecords.push({
+        type : 'TXT',
+        name : `_${dnsRecordId}._did.`,
+        ttl  : DNS_RECORD_TTL,
+        data : txtData.join(PROPERTY_SEPARATOR)
+      });
+    });
+
+    // Initialize the root DNS TXT record with the DID DHT specification version.
+    const rootRecord: string[] = [`v=${DID_DHT_SPECIFICATION_VERSION}`];
+
+    // Add verification methods to the root record.
+    if (verificationMethodIds.length) {
+      rootRecord.push(`vm=${verificationMethodIds.join(VALUE_SEPARATOR)}`);
+    }
+
+    // Add verification relationships to the root record.
+    Object.keys(DidVerificationRelationship).forEach(relationship => {
+      // Collect the verification method IDs for the given relationship.
+      const dnsRecordIds = (didDocument[relationship as keyof DidDocument] as any[])
+        ?.map(id => idLookup.get(id.split('#').pop()));
+
+      // If the relationship includes verification methods, add them to the root record.
+      if (dnsRecordIds) {
+        const recordName = DidDhtVerificationRelationship[relationship as keyof typeof DidDhtVerificationRelationship];
+        rootRecord.push(`${recordName}=${dnsRecordIds.join(VALUE_SEPARATOR)}`);
+      }
+    });
+
+    // Add services to the root record.
+    if (serviceIds.length) {
+      rootRecord.push(`svc=${serviceIds.join(VALUE_SEPARATOR)}`);
+    }
+
+    // If defined, add a DNS TXT record for each registered DID type.
+    if (didMetadata.types?.length) {
+      // DID types can be specified as either a string or a number, so we need to normalize the
+      // values to integers.
+      const types = didMetadata.types as (DidDhtRegisteredDidType | keyof typeof DidDhtRegisteredDidType)[];
+      const typeIntegers = types.map(type => typeof type === 'string' ? DidDhtRegisteredDidType[type] : type);
+
+      dnsAnswerRecords.push({
+        type : 'TXT',
+        name : '_typ._did.',
+        ttl  : DNS_RECORD_TTL,
+        data : `id=${typeIntegers.join(VALUE_SEPARATOR)}`
+      });
+    }
+
+    // Add a DNS TXT record for the root record.
+    dnsAnswerRecords.push({
```

Comment:
Does this need to be in an else statment? Otherwise you can write 2 txt records.

Is that expected?

Reply:
I might be missing the issue or question.  There are potentially many TXT records for a given `did:dht` DID (alsoKnownAs, controller, one for each verification method, one for each service, etc.).

The line you highlighted is adding the final root record, of which there is only one.  However, it is a required record so every DHT DID will have at least 1 TXT record for the root record.